### PR TITLE
Remove StoredSubject and UserDefault wrapper

### LINKED ISSUE #42: Deprecate UserDefault property wrapper on iOS 14+
Prefer usage of `AppStorage` on iOS 14+.

### PR DESCRIPTION
**Pull request by mkj-is**
Closes #42.

`@UserDefaults` was removed in favor of  [`@AppStorage`](https://developer.apple.com/documentation/swiftui/appstorage/) in iOS 14+.
`@StoredSubject` was removed in favor of  [`CurrentValueSubject`](https://developer.apple.com/documentation/combine/currentvaluesubject) in iOS 13+.

## Discussion

We took the radical approach so we didn't have to add all the available attributes to all the tiny types.

```swift
@available(iOS, deprecated: 14.0, message: "Deprecated in favor of AppStorage")
@available(tvOS, deprecated: 14.0, message: "Deprecated in favor of AppStorage")
@available(macOS, deprecated: 11.0, message: "Deprecated in favor of AppStorage")
@available(watchOS, deprecated: 7.0, message: "Deprecated in favor of AppStorage")
```

### CODE REVIEW

File: Tests/FTPropertyWrappersTests/XCTestManifests.swift [hunk 1]
```diff
@@ -5,8 +5,6 @@ public func allTests() -> [XCTestCaseEntry] {
     return [
         testCase(KeychainTests.allTests),
         testCase(SerializedTests.allTests),
-        testCase(UserDefaultsTests.allTests),
-        testCase(StoredSubjectTests.allTests),
```
Comment (marieRe18) [inline Tests/FTPropertyWrappersTests/XCTestManifests.swift:9]: Jestli jsem dobre pochopila, tak se zmenil pristup, jak k temto objektu pristupovat -> nemely by se i presto nejake testy zachovat/napsat nove?

File: README.md [hunk 2]
```diff
@@ -23,84 +23,11 @@ pod 'FTPropertyWrappers', '~> 1.0'
 
 The main aim of this package is to provide programmer with access to commonly used features or snippets with as easy API as possible. Runtime efficiency, while important, is not the main focus of this package. As for today, this package contains wrappers for following features:
 
-- `UserDefaults` for storing a value inside Foundation's User Defaults
-- `StoredSubject` which is simple multidelegate/observing primitive for Swift
```
Comment (marieRe18) [inline README.md:27]: Kdyz tyto objekty byly nahrazeny, nemely byt zde zmineny ty nove tedy?

Comment (mkj-is) [reply README.md:27]: Nene, `CurrentValueSubject` je přímo součástí `Combine` a `AppStorage` zase `SwiftUI`.

File: Tests/FTPropertyWrappersTests/XCTestManifests.swift (hunk 1 above)
Comment (mkj-is) [reply Tests/FTPropertyWrappersTests/XCTestManifests.swift:9]: Na systémové věci, které jsou mimo scope této package úplně nevím jaké testy bychom psali.

File: README.md (hunk 2 above)
Comment (marieRe18) [reply README.md:27]: Aha, pak jsem asi nepochopila, proc se tu zminovalo UserDefaults :D to je soucasti Foundation, coz mi prijde taky jako zaklad

**Review by tomasbabulak — DISMISSED** [review]

File: FTPropertyWrappers.podspec [hunk 3]
```diff
@@ -1,10 +1,9 @@
 Pod::Spec.new do |s|
   s.name         = "FTPropertyWrappers"
-  s.version      = "1.1.0"
+  s.version      = "1.2.0"
```
Comment (mikolasstuchlik) [inline FTPropertyWrappers.podspec:3]: The API introduced in this PR is incompatible with the old one. 

> MAJOR version when  you make incompatible API changes,
> MINOR version when you add functionality in a backwards compatible manner, and
> PATCH version when you make backwards compatible bug fixes.
> Additional labels for pre-release and build metadata are available as extensions to the MAJOR.MINOR.PATCH format.
https://semver.org

We should release this as a version 2.0.0.

**Review by mikolasstuchlik — CHANGES_REQUESTED** [review]

File: FTPropertyWrappers.podspec (hunk 3 above)
Comment (mikolasstuchlik) [reply FTPropertyWrappers.podspec:3]: I would also ask the team for ideas on more complex property wrappers that should be included in this repo. I'm not talking about things like `Clamping` but something more challenging (like the Keychain is).

I am on the fence about having 1 property wrapper in a repo called FRPropertyWrapper`S`.

Comment (mkj-is) [reply FTPropertyWrappers.podspec:3]: Sure, let's release it as major version.

```suggestion
  s.version      = "2.0.0"
```

I agree that the name did not age well. But I am strongly against forcefully adding new property wrappers only to name makes sense. There are still two useful property wrappers remaining – I know they are both related to keychain, but I am not sure it's reasanable to rename the package.

File: README.md [hunk 4]
```diff
@@ -12,7 +12,7 @@ When using Swift package manager install using Xcode 11+
 or add following line to your dependencies:
 
 ```swift
-.package(url: "https://github.com/futuredapp/FTPropertyWrappers.git", from: "1.0.0")
+.package(url: "https://github.com/futuredapp/FTPropertyWrappers.git", from: "1.2.0")
```
Comment (mkj-is) [inline README.md:15]:
```suggestion
.package(url: "https://github.com/futuredapp/FTPropertyWrappers.git", from: "2.0.0")
```

File: README.md [hunk 5]
```diff
@@ -241,7 +168,13 @@ It appears, that in case of ambiguity, element with the oldest `creationDate` is
 
 ## Migration notes
 
-### 0.2.0 ~> 1.0.0
+### 1.2.0
```
Comment (mikolasstuchlik) [inline README.md:171]:
```suggestion
### 2.0.0
```

**Review by mikolasstuchlik — APPROVED** [review]

**Pull request merged**